src/App.tsx

```diff
@@ -3,24 +3,48 @@ import { Navbar } from 'reactstrap';
 
 import { Footer } from './components/Footer';
 import { WebMapContainer } from './containers/WebMapContainer';
+import { IdentityContainer } from './containers/IdentityContainer';
 
 import './App.css';
 
-class App extends React.Component<object, object> {
+const portalUrl = 'https://www.arcgis.com/sharing';
+
+interface AppProps {}
+interface AppState {
+  credential?: __esri.Credential;
+}
+
+class App extends React.Component<AppProps, AppState> {
+  constructor(props: AppProps) {
+    super(props);
+    this.state = {};
+    this.handleLoginChange = this.handleLoginChange.bind(this);
+  }
+
+  handleLoginChange(credential?: __esri.Credential) {
+    this.setState({credential});
+  }
```

Comment:
I went back and forth in my code whether to do `handleLoginChange` vs `login` and `logout`. Basically, should we have one method to call both the 'on' and 'off' states or to separate them into two. Two methods seems a bit clearer but is a bit more boilerplate.

Perhaps we should agree that one method toggling the states is our way of doing it so as to keep code short.  

Maybe I'm overthinking this...

Reply:
I think you're overthinking this. Both approaches are valid, I don't one is superior to the other one. If you insist, I can change it but I'll leave it for now...